client/src/main/java/io/split/client/SplitFactoryImpl.java

```diff
@@ -80,14 +87,31 @@ public SplitFactoryImpl(String apiToken, SplitClientConfig config) throws IOExce
         cm.setMaxTotal(20);
         cm.setDefaultMaxPerRoute(20);
 
-        final CloseableHttpClient httpclient = HttpClients.custom()
+        HttpClientBuilder httpClientbuilder = HttpClients.custom()
                 .setConnectionManager(cm)
                 .setDefaultRequestConfig(requestConfig)
                 .setSSLSocketFactory(sslsf)
                 .addInterceptorLast(AddSplitHeadersFilter.instance(apiToken))
                 .addInterceptorLast(new GzipEncoderRequestInterceptor())
-                .addInterceptorLast(new GzipDecoderResponseInterceptor())
-                .build();
+                .addInterceptorLast(new GzipDecoderResponseInterceptor());
+
+        // Set up proxy is it exists
+        if (config.proxy() != null) {
+            _log.info("Initializing Split SDK with proxy settings");
+            DefaultProxyRoutePlanner routePlanner = new DefaultProxyRoutePlanner(config.proxy());
+            httpClientbuilder.setRoutePlanner(routePlanner);
+
+            if (config.proxyUsername() != null && config.proxyPassword() != null){
+                CredentialsProvider credsProvider = new BasicCredentialsProvider();
```

Comment:
Should we log (in debug) something like "Initializaing proxy with user: userr and password: ***** 
just to make sure for debugging purposes to know that we are setting user/pass